### PR TITLE
[WEB-2380] chore: cycle sidebar refactor

### PR DESCRIPTION
**Pull request by anmolsinghbhatia**
### Summary
Cycle sidebar code refactoring

[[WEB-2380]](https://app.plane.so/plane/projects/02c3e1d5-d7e2-401d-a773-45ecba45d745/issues/5567b9d4-0f31-465b-8021-427b7c557e19)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `SidebarChart` component for visualizing progress data.
	- Added a `SidebarChartRoot` component to wrap the chart functionality.
	- Added a progress percentage display in the `CycleAnalyticsProgress` component.

- **Improvements**
	- Enhanced cycle details rendering for better clarity and organization.
	- Improved date validation logic in the `CycleSidebarHeader` component to prevent potential errors.

These updates enhance user experience by providing clearer visualizations and more robust functionality.

### CODE REVIEW

File: web/core/components/cycles/analytics-sidebar/issue-progress.tsx [hunk 1]
```diff
@@ -17,9 +17,8 @@ import { EIssueFilterType, EIssuesStoreType } from "@/constants/issue";
 import { getDate } from "@/helpers/date-time.helper";
 // hooks
 import { useIssues, useCycle, useProjectEstimates } from "@/hooks/store";
-// plane web constants
-import { SidebarBaseChart } from "@/plane-web/components/cycles/analytics-sidebar";
 import { EEstimateSystem } from "@/plane-web/constants/estimates";
+import SidebarChart from "./sidebar-chart";
```
Comment (rahulramesha) [inline web/core/components/cycles/analytics-sidebar/issue-progress.tsx:21]: Previously it was coming from Plane-web. Is this intentional and wont it cause any issues in EE?

File: web/core/components/cycles/analytics-sidebar/sidebar-header.tsx [hunk 2]
```diff
@@ -156,7 +156,7 @@ export const CycleSidebarHeader: FC<Props> = observer((props) => {
       end_date: renderFormattedPayloadDate(endDate),
     };
 
-    if (cycleDetails?.start_date && cycleDetails.end_date)
+    if (cycleDetails && cycleDetails.start_date && cycleDetails.end_date)
```
Comment (rahulramesha) [inline web/core/components/cycles/analytics-sidebar/sidebar-header.tsx:159]: Any reason for this change? both looks same

Comment (anmolsinghbhatia) [reply web/core/components/cycles/analytics-sidebar/sidebar-header.tsx]: I've reverted the change.

File: web/core/components/cycles/analytics-sidebar/issue-progress.tsx (hunk 1 above)
Comment (anmolsinghbhatia) [reply web/core/components/cycles/analytics-sidebar/issue-progress.tsx]: Resolved

**Review by rahulramesha — APPROVED** [review]

**Pull request merged**